src/mjolnir/adminbuilder.cc

```diff
@@ -12,206 +12,140 @@
 
 #include "config.h"
 
-/* Need to know which geos version we have to work out which headers to include */
-#include <geos/version.h>
-
-#define USE_UNSTABLE_GEOS_CPP_API
-#if GEOS_VERSION_MAJOR == 3 && GEOS_VERSION_MINOR >= 8
-#include <geos/geom/CoordinateArraySequence.h>
-#else
-#include <geos/geom/CoordinateSequenceFactory.h>
-#endif
-#include <geos/geom/Geometry.h>
-#include <geos/geom/GeometryFactory.h>
-#include <geos/geom/LineString.h>
-#include <geos/geom/LinearRing.h>
-#include <geos/geom/MultiLineString.h>
-#include <geos/geom/MultiPolygon.h>
-#include <geos/geom/Point.h>
-#include <geos/geom/Polygon.h>
-#include <geos/io/WKTReader.h>
-#include <geos/io/WKTWriter.h>
-#if GEOS_VERSION_MAJOR == 3 && GEOS_VERSION_MINOR >= 9
-#include <geos/operation/linemerge/LineMerger.h>
-#else
-#include <geos/opLinemerge.h>
-#endif
-#include <geos/util/GEOSException.h>
-
-#include <boost/optional.hpp>
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
 #include <boost/property_tree/ptree.hpp>
 
-using namespace geos::geom;
-using namespace geos::io;
-using namespace geos::util;
-using namespace geos::operation::linemerge;
+using point_t =
+    boost::geometry::model::d2::point_xy<double,
+                                         boost::geometry::cs::geographic<boost::geometry::degree>>;
+using polygon_t = boost::geometry::model::polygon<point_t>;
+using multipolygon_t = boost::geometry::model::multi_polygon<polygon_t>;
+
+namespace std {
+template <> struct hash<point_t> {
+  inline size_t operator()(const point_t& p) const {
+    return uint64_t(p.x() * 1e7 + 180 * 1e7) << 32 | uint64_t(p.y() * 1e7 + 90 * 1e7);
+  }
+};
+
+template <> struct equal_to<point_t> {
+  inline bool operator()(const point_t& a, const point_t& b) const {
+    return a.x() == b.x() && a.y() == b.y();
+  }
+};
+
+} // namespace std
 
 // For OSM pbf reader
 using namespace valhalla::mjolnir;
 using namespace valhalla::baldr;
-
 using namespace valhalla::midgard;
 
 namespace {
 
+/**
+ * Converts a series of a linestrings into one or more polygons (rings) by connecting contiguous
+ * ones until a ring is formed
+ * @param lines the unmerged lines
+ * @param line_lookup a multi map that lets one easily
+ * @return
+ */
+std::vector<std::vector<point_t>> line_merge(std::vector<std::vector<point_t>> lines,
+                                             std::unordered_multimap<point_t, size_t> line_lookup) {
+  // TODO: use the lookup to connect adjacent linestrings. GriddedData::GenerateContours does this
```

Comment:
this is the only thing left to stub in before i can really test the code. ive written this algorithm before over in the isochrones stuff but i'm not sure if i'll templatize it and try to reuse it or just reimpliment the spirit of it. the start i have here already assumes a slightly different approach so i might try that as a possible improvement on the one that uses ordered containers. at the same time i might be forgetting a gotcha that a younger more intelligent me knew :D